test/boost/mutation_test.cc

```diff
@@ -1980,9 +1980,19 @@ SEASTAR_THREAD_TEST_CASE(test_cell_equals) {
     BOOST_REQUIRE(!c4.equals(*bytes_type, c1));
 }
 
+// Global to avoid elimination by the compiler; see below for use
+thread_local data_type force_type_thread_local_init_evaluation [[gnu::used]];
+
 SEASTAR_THREAD_TEST_CASE(test_cell_external_memory_usage) {
     measuring_allocator alloc;
 
+    // Force evaluation of int33_type and all the other types. This is so
+    // the compiler doesn't reorder it into the with_allocator section, below,
+    // and any managed_bytes instances creates during that operation interfere
+    // with the measurements.
+    force_type_thread_local_init_evaluation = int32_type;
```

Comment:
just curious, wouldn't a simple
```c++
(void) int2_type;
```
work too? Is that variable really necessary just to avoid optimization?

Reply:
The compiler can optimize your code to nothing.

The variable isn't needed to avoid optimization, it's needed to avoid code motion. The sequence "assign to global variable; signal fence" cannot be reordered, since the compiler can't be sure a signal won't examine the variable after the fence.